auth/src/test/java/io/grpc/auth/GoogleAuthLibraryCallCredentialsTests.java

```diff
@@ -269,6 +274,55 @@ public void serviceUri() throws Exception {
     verify(credentials).getRequestMetadata(eq(new URI("https://example.com:123/a.service")));
   }
 
+  @Test
+  public void serviceAccountToJwt() throws Exception {
+    KeyPair pair = KeyPairGenerator.getInstance("RSA").generateKeyPair();
+    ServiceAccountCredentials credentials = new ServiceAccountCredentials(
+        null, "email@example.com", pair.getPrivate(), null, null) {
+      @Override
+      public AccessToken refreshAccessToken() {
+        throw new AssertionError();
+      }
+    };
+
+    GoogleAuthLibraryCallCredentials callCredentials =
+        new GoogleAuthLibraryCallCredentials(credentials);
+    callCredentials.applyRequestMetadata(method, attrs, executor, applier);
+    assertEquals(1, runPendingRunnables());
+
+    verify(applier).apply(headersCaptor.capture());
+    Metadata headers = headersCaptor.getValue();
+    String[] authorization = Iterables.toArray(headers.getAll(AUTHORIZATION), String.class);
+    assertEquals(1, authorization.length);
+    assertTrue(authorization[0], authorization[0].startsWith("Bearer "));
+    // JWT is reasonably long. Normal tokens aren't.
+    assertTrue(authorization[0], authorization[0].length() > 300);
+  }
+
+  @Test
+  public void serviceAccountWithScopeNotToJwt() throws Exception {
+    final AccessToken token = new AccessToken("allyourbase", new Date(Long.MAX_VALUE));
+    KeyPair pair = KeyPairGenerator.getInstance("RSA").generateKeyPair();
+    ServiceAccountCredentials credentials = new ServiceAccountCredentials(
+        null, "email@example.com", pair.getPrivate(), null, Arrays.asList("somescope")) {
+      @Override
+      public AccessToken refreshAccessToken() {
+        return token;
+      }
+    };
+
+    GoogleAuthLibraryCallCredentials callCredentials =
+        new GoogleAuthLibraryCallCredentials(credentials);
+    callCredentials.applyRequestMetadata(method, attrs, executor, applier);
+    assertEquals(1, runPendingRunnables());
+
+    verify(applier).apply(headersCaptor.capture());
+    Metadata headers = headersCaptor.getValue();
+    Iterable<String> authorization = headers.getAll(AUTHORIZATION);
+    assertArrayEquals(new String[]{"Bearer allyourbase"},
```

Comment:
Optional: Truth has some nice fluent overloads for asserting about arrays/collections.